ci/views.py

```diff
@@ -493,6 +492,216 @@ def identities(request):
+                    settings.IDENTITY_FIELD: identity
+                })
+
+                messages.add_message(
+                    request,
+                    messages.INFO,
+                    'Successfully opted out.',
+                    extra_tags='success'
+                )
+            except:
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    'Optout failed.',
+                    extra_tags='danger'
+                )
+
+    hub_filter = {
+        settings.IDENTITY_FIELD: identity
+    }
+    registrations = hubApi.get_registrations(params=hub_filter)
+    changes = hubApi.get_changes(params=hub_filter)
+    if results is None:
+        return redirect('not_found')
+
+    outbound_message_params = {
+        'to_identity': identity,
+        'ordering': '-created_at',
+    }
+    outbound_messages = msApi.get_outbounds(params=outbound_message_params)
+    outbound_page = request.GET.get('outbound_page')
+    outbound_paginator = Paginator(
+        list(outbound_messages['results']),
+        settings.IDENTITY_MESSAGES_PAGE_SIZE)
+
+    try:
+        outbound_messages = outbound_paginator.page(outbound_page)
+    except PageNotAnInteger:
+        outbound_messages = outbound_paginator.page(1)
+    except EmptyPage:
+        outbound_messages = outbound_paginator.page(
+            outbound_paginator.num_pages)
+
+    inbound_message_params = {
+        'from_identity': identity,
+        'ordering': '-created_at',
+    }
+    inbound_messages = msApi.get_inbounds(inbound_message_params)
+    inbound_page = request.GET.get('inbound_page')
+    inbound_paginator = Paginator(
+        list(inbound_messages['results']),
+        settings.IDENTITY_MESSAGES_PAGE_SIZE)
+
+    try:
+        inbound_messages = inbound_paginator.page(inbound_page)
+    except PageNotAnInteger:
+        inbound_messages = inbound_paginator.page(1)
+    except EmptyPage:
+        inbound_messages = inbound_paginator.page(inbound_paginator.num_pages)
+
+    deactivate_subscription_form = DeactivateSubscriptionForm()
+    add_subscription_form = AddSubscriptionForm()
+    add_subscription_form.fields['messageset'] = forms.ChoiceField(
+                                                    choices=choices)
+
+    optout_visible = False
+    details = results.get('details', {})
+    addresses = details.get('addresses', {})
+    msisdns = addresses.get('msisdn', {})
+    optout_visible = any(
+        (not d.get('optedout') for _, d in msisdns.items()))
+    
+   
+
+    if results['details'].get('linked_to'):
+        linked_to = idApi.get_identity(results['details']['linked_to'])
+
+    print (results)
+    if results['details'].get('operator', results.get('operator')):
+        print ("hellllllllooooooo operator", operator_id)
+        operator_id = idApi.get_identity(results['details'].get('operator', results.get('operator')))
+         
+    
+    context = {
```

Comment:
Please remove anything in here that is not used in the template. I think we'll be able to remove a bunch of code above too. For example `inbound_messages`, it's not used in the template so we can remove it here and above where we do the api lookup.